core/embed/rust/src/zcash/diag.rs

```diff
@@ -0,0 +1,90 @@
+use crate::micropython::{buffer::Buffer, obj::Obj};
+use crate::util;
+use core::{convert::TryFrom, ops::Deref};
+use group::GroupEncoding;
+use orchard::keys::{
+    FullViewingKey, NullifierDerivingKey, SpendAuthorizingKey, SpendValidatingKey, SpendingKey,
+};
+use orchard::primitives::sinsemilla::{CommitDomain, HashDomain};
+
+pub const COMMIT_IVK_PERSONALIZATION: &str = "z.cash:Orchard-CommitIvk";
+
+fn as_u32_le(array: &[u8]) -> u32 {
```

Comment:
I know you've mentioned this file is not up for review, but a small note: `&[u8]` impl `TryInto<[u8; 4]>`, so you can write this as `u32::from_le_bytes(array.try_into().unwrap())`. Likewise for the lower fn.

Reply:
Nice. Thanks.